job instead of task config

bdevloed opened the pull request:
Config by job operation instead of task.

To be used together with https://github.com/lblod/app-verenigingen-loket-harvester/pull/25

bdevloed committed:
job instead of task config

bdevloed committed:
make deletion safer

bdevloed committed:
Update readme

benjay10 commented on lib/queries.js:
```diff
@@ -3,178 +3,104 @@ import { querySudo as query, updateSudo as update } from "@lblod/mu-auth-sudo";
 };
+// This function is no longer needed as we're not using dump file logic
+// Keeping it for backward compatibility but it returns a far past date
 export async function getLastDumpFileJobDate() {
```
If it's no longer used anywhere in the code, why keep this deprecated function?

Like this, it no longer has any useful body, so it also does not even serve as a 'comment' or 'suggestion'. The previous implementations of this function still remains in the git history.

benjay10 commented on lib/pipeline-cleanup.js:
```diff
@@ -12,55 +12,64 @@ import {
+    // 2. Busy jobs older than MAX_DAYS_TO_KEEP_BUSY_JOBS
+    // 3. Failed jobs older than MAX_DAYS_TO_KEEP_FAILED_JOBS
+    const jobsToClean = [
+      ...(await getSuccessfulJobsBeforeDate()),
```
I feels strange to not give a date as argument like you do with the other calls below.

benjay10 commented on lib/queries.js:
```diff
@@ -3,178 +3,104 @@ import { querySudo as query, updateSudo as update } from "@lblod/mu-auth-sudo";
+export async function getSuccessfulJobsBeforeDate() {
+  const maxDaysToKeepSuccessfulJobs = new Date();
+  maxDaysToKeepSuccessfulJobs.setDate(
+    maxDaysToKeepSuccessfulJobs.getDate() - MAX_DAYS_TO_KEEP_SUCCESSFUL_JOBS,
```
I feel like this date calculation should be done when calling the function, so not in this `queries.js` file, where we expect to see only (or mostly only) queries and supporting code for executing those queries.

benjay10 committed:
Whitespaces

benjay10 committed:
Outlining and capitalizing in queries

benjay10 committed:
Linting

benjay10 committed:
Remove unused packages

benjay10 committed:
Remove dependency on lodash

benjay10 committed:
Commit package-lock

benjay10 committed:
Bump mu-javascript-template

benjay10 committed:
Use built in json import, fail on error

benjay10 committed:
Remove unneeded comment

benjay10 committed:
Use PREFIXES

benjay10 committed:
Minor query formatting changes

benjay10 committed:
Remove unused functions

benjay10 committed:
Unify 2 similar functions

benjay10 committed:
Rename function to signify callback behaviour better

benjay10 committed:
Spelling mistake

benjay10 committed:
README: corrected API and spacing

benjay10 committed:
Install and use env-var

This resolves a problem where environment variables have the wrong type
(e.g. numbers as strings due to the nature of environment variables).

benjay10 committed:
Fix wrong prefix for DataContainer

benjay10 reviewed (changes requested):
I would like to change a few things. Already have a slew of changes prepared, but cannot commit them because I don't have permissions yet.

benjay10 commented:
I know I changed a lot of things, mostly just for cleaning up the code, or to simplify things. I tested this code with a debugger on Incremental Collecting and full Collecting tasks from the Verenigingen Harvester and this seemed to work correctly.

The only actual bug I found was fixed in commit: https://github.com/lblod/harvesting-cleanup-previous-jobs-service/pull/4/commits/6bbc3c758cd0413cbc14c9b5879d452decc76d56
